app/org/sagebionetworks/bridge/models/subpopulations/SubpopulationGuidImpl.java

```diff
@@ -0,0 +1,36 @@
+package org.sagebionetworks.bridge.models.subpopulations;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Objects;
+
+public final class SubpopulationGuidImpl implements SubpopulationGuid {
```

Comment:
I really dislike this pattern. How much work are we actually saving by allowing Subpops and SubpopGuids to be used interchangeably? In contrast, using this pattern means we have an extra interface, and this interface is sometimes mutable and sometimes immutable, and if Subpops and SubpopGuids are interchangeable, we actually *lose* a bit of type-safety.

Given that this pattern has already caused a lot of confusion for studies and surveys, I'd strongly prefer to not use this pattern here.